integration_test/tests/odbc_2_tests.rs

```diff
@@ -0,0 +1,318 @@
+        // Allocate a DBC handle
+        let mut dbc: Handle = null_mut();
+        #[allow(unused_mut)]
+        let mut output_len;
+        let in_connection_string;
+        let out_connection_string;
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Dbc,
+                    env_handle as *mut _,
+                    &mut dbc as *mut Handle
+                )
+            );
+
+            // Causes iODBC to hang on `SetConnectOptionW` call
+            #[cfg(not(target_os = "macos"))]
+            {
+                // Set the login timeout
+                let login_timeout = 15;
+                assert_eq!(
+                    SqlReturn::SUCCESS,
+                    SQLSetConnectAttrW(
+                        dbc as HDbc,
+                        ConnectionAttribute::LoginTimeout,
+                        login_timeout as Pointer,
+                        0,
+                    )
+                );
+            }
+
+            in_connection_string = generate_default_connection_str();
+            let in_connection_string_encoded = cstr::to_widechar_vec(&in_connection_string);
+
+            let str_len_ptr = &mut 0;
+            const BUFFER_LENGTH: SmallInt = 300;
+            let mut out_connection_string_buff: [WideChar; BUFFER_LENGTH as usize - 1] =
+                [0; (BUFFER_LENGTH as usize - 1)];
+            let out_connection_string_buff = &mut out_connection_string_buff as *mut WideChar;
+
+            assert_ne!(
+                SqlReturn::ERROR,
+                SQLDriverConnectW(
+                    dbc as HDbc,
+                    null_mut(),
+                    in_connection_string_encoded.as_ptr(),
+                    NTS as SmallInt,
+                    out_connection_string_buff,
+                    BUFFER_LENGTH,
+                    str_len_ptr,
+                    DriverConnectOption::NoPrompt,
+                ),
+                "{}",
+                get_sql_diagnostics(HandleType::Dbc, dbc)
+            );
+
+            output_len = *str_len_ptr;
+            // The iodbc driver manager is multiplying the output length by size_of WideChar (u32)
+            // for some reason. It is correct when returned from SQLDriverConnectW, but is 4x
+            // bigger between return and here.
+            if odbc_sys::USING_IODBC {
+                output_len /= std::mem::size_of::<WideChar>() as i16;
+            }
+
+            out_connection_string = cstr::from_widechar_ref_lossy(slice::from_raw_parts(
+                out_connection_string_buff,
+                output_len as usize,
+            ));
+        }
+        (
+            dbc as HDbc,
+            in_connection_string,
+            out_connection_string,
+            output_len,
+        )
+    }
+
+    /// Test Setup flow
+    #[test]
+    fn test_setup() {
+        setup();
+    }
+
+    /// Test connect
+    #[test]
+    fn test_connect() {
```

Comment:
This was just testing that creating the connection handle doesn't panic basically. But I agree, maybe we can merge `test_setup` into this because they are so overlapping